cattle/plugins/docker/compute.py

```diff
@@ -201,68 +249,161 @@ def _setup_ports(create_config, instance):
         if len(ports) > 0:
             create_config['ports'] = ports
 
-    def _do_instance_activate(self, instance, host, progress):
-        def to_upper_case(key):
-            return key[0].upper() + key[1:]
+    def _record_rancher_container_state(self, client, instance,
+                                        docker_id=None):
+        if docker_id is None:
+            container = self.get_container(client, instance)
+            if container is not None:
+                docker_id = container['Id']
 
-        name = instance.uuid
-        try:
-            image_tag = instance.image.data.dockerImage.fullName
-        except KeyError:
-            raise Exception('Can not start container with no image')
+        if docker_id is None:
+            return
 
-        c = self._get_docker_client(host)
+        cont_dir = Config.container_state_dir()
 
-        create_config = {
-            'name': name,
-            'detach': True
-        }
+        file_path = path.join(cont_dir, docker_id)
+        if os.path.exists(file_path):
+            os.remove(file_path)
+
+        if not os.path.exists(cont_dir):
+            os.makedirs(cont_dir)
+
+        with open(file_path, 'w') as outfile:
```

Comment:
No, its creation is picked up by host-api which uses it to inject the rancher ip into the container.